spring-data-cassandra/src/main/java/org/springframework/data/cassandra/core/mapping/StaticColumn.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2013-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.cassandra.core.mapping;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Annotation to mark a field static.
+ *
+ * @author Aleksei Zotov
+ */
+@Documented
+@Retention(RetentionPolicy.RUNTIME)
+@Target(value = { ElementType.ANNOTATION_TYPE, ElementType.FIELD, ElementType.METHOD })
+public @interface StaticColumn {
```

Comment:
Instead of introducing another annotation type, it seems that we should rather extend `@Column` and probably `@PrimaryKeyColumn` by adding a `staticColumn` or `isStatic` attribute.

Reply:
As proposed, I've added `isStatic` attribute to `@Column` annotation. There is no need to add that to `@PrimaryKeyColumn`  annotation because primary columns (partitioned/clustered) cannot be static ([reference to documentation](https://docs.datastax.com/en/dse/6.8/cql/cql/cql_using/refStaticCol.html)). There is a separate check for that:
```
Assert.isTrue(!(optionalKeyType.isPresent() && isStatic),"PrimaryKey must not be static");
```